mod-users/src/main/resources/domain/mod-users/features/users.feature

```diff
@@ -0,0 +1,111 @@
+    * def responseBarcode = createUserResponse.response.firstName
+    Given path 'users?query=(personal.firstName=first)'
+    When method GET
+    Then status 200
+    And match response.users[0].personal.firstName == expectedFirstName
+
+  Scenario: Search user by firstname & lastname.
+    * def expectedFirstName = 'first'
+    * def expectedLastName = 'TestUser'
+    * def createUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser')
+    * def responseBarcode = createUserResponse.response.firstName
+    Given path 'users?query=(personal.firstName=first)or(personal.lastName=TestUser)'
+    When method GET
+    Then status 200
+    And match response.users[0].personal.firstName == expectedFirstName
+    And match response.users[0].personal.lastName == expectedLastName
+
+  Scenario: Search user by UUID.
+    * def createUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser')
+    * def responseUserId = createUserResponse.response.id
+
+    Given path 'users?query=(id='+responseUserId+')'
+    When method GET
+    Then status 200
+    And match responseUserId == response.users[0].id
+
+  Scenario: Search user by lastname.
+    * def expectedLastName = 'TestUser'
+    * def createUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser')
+    * def responseUserLastName = createUserResponse.response.lastName
+
+    Given path 'users?query=(personal.lastName=TestUser)'
+    When method GET
+    Then status 200
+    And match responseUserLastName == response.users[0].lastName
+
+  Scenario: Search user by email.
+    * def expectedEmail = 'testmail@abc.com'
+    * def createUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser')
+    * def responseUserEmail = createUserResponse.response.email
+
+    Given path 'users?query=(personal.email=testmail@abc.com)'
+    When method GET
+    Then status 200
+    And match responseUserEmail == response.users[0].email
+
+  Scenario: Search user by username.
+    * def expectedEmail = 'testmail@abc.com'
+    * def createUserResponse = call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser')
+    * def responseUserEmail = createUserResponse.response.email
+
+    Given path 'users?query=(username=test-user)'
+    When method GET
+    Then status 200
+
+  Scenario: Filter inactive patron.
+    * call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser') { status: true }
+    * call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser') { status: true }
+    * call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser') { status: false }
+    * call read('classpath:domain/mod-users/features/util/initData.feature@PostPatronGroupAndUser') { status: false }
```

Comment:
This filter scenario defines both matching and non-matching data to demonstrate the criteria is applied as expected.

The search scenarios don't seem to do this. They only define the matching data.

Please can you help me understand why this is done for this scenario and not the others?

Reply:
Yes i agree, 
i think in search scenario both matching and non-matching criteria should be there. i will correct it.  thankyou